yt_dlp/YoutubeDL.py

```diff
@@ -911,10 +911,8 @@ def report_warning(self, message, only_once=False):
             self.to_stderr(f'{self._format_err("WARNING:", self.Styles.WARNING)} {message}', only_once)
 
     def deprecation_warning(self, message):
-        if self.params.get('logger') is not None:
-            self.params['logger'].warning('DeprecationWarning: {message}')
-        else:
-            self.to_stderr(f'{self._format_err("DeprecationWarning:", self.Styles.ERROR)} {message}', True)
+        import warnings
```

Comment:
any reason why the import is not at module level?

Also, I am not sure how warnings module works with the windows exe. There are a couple of places where the deprecation warnings are user-facing (as opposed to API only). So I'll need to test that

Reply:
Optimizing for the happy path, where this never gets invoked? Also, putting imports at the top creates the potential for merge conflicts.

Do we catch deprecation warnings in CI? I think at least the 3.10 run should have warnings converted into errors.